irohad/main/application.cpp

```diff
@@ -103,7 +105,10 @@ void Irohad::dropStorage() {
  * Initializing iroha daemon storage
  */
 void Irohad::initStorage() {
-  auto storageResult = StorageImpl::create(block_store_dir_, pg_conn_);
+  auto factory =
+      std::make_shared<shared_model::proto::ProtoCommonObjectsFactory<
+          shared_model::validation::FieldValidator>>();
+  auto storageResult = StorageImpl::create(block_store_dir_, pg_conn_, factory);
```

Comment:
Maybe create ```factory``` variable directly inside ```StorageImpl::create(..)```? It's not used in code, and the space will be saved

Reply:
We want to isolate protobuf specific implementations from the rest of the system.